controllers/replica_calculator.go

```diff
@@ -75,6 +74,12 @@ func (c *ReplicaCalculator) GetExternalMetricReplicas(logger logr.Logger, target
 		return ReplicaCalculation{}, fmt.Errorf("unable to get the number of ready pods across all namespaces for %v: %s", lbl, err.Error())
 	}
 
+	if currentReadyReplicas > target.Status.Replicas {
+		// if we enter in that condition it means that several Deployment use the same label selector in the same namespace.
+		logger.Info("The number of current ready Pods is higher that the current status.replicas", "currentReadyReplicas", currentReadyReplicas, "targetStatusReplicas", target.Status.Replicas)
```

Comment:
Maybe the log message itself should contain the information that the most probable cause is a bad label selector, which is a user misconfiguration.
Written as it is, the log isn’t explicit about the fact that there is something to be fixed.
Also, the log misses the needed information to identify the guilty WPA: its name and namespace.

Reply:
the `logger` inherit automatically from the ns/name of the current WPA, not need to add to it on each `logger.Info()` call that the beauty of thy logger :) 

something like
```golang
logger = logger.WithValue("wpa", wpa.Namespace + "/" + wpa.Name)
```
is made in the main `Reconcile()` function

Reply:
For the message I agree, I can be more explicit.